src/display_noritake.c

```diff
@@ -0,0 +1,208 @@
+#include "CompDevSDK.h"
+
+#ifdef DISPLAY_NORITAKE
+
+#ifndef NORITAKE_SIO
+#warning NORITAKE_SIO not defined, pin 10 will be used
+#define NORITAKE_SIO 10
+#endif
+#ifndef NORITAKE_STB
+#warning NORITAKE_STB not defined, pin 11 will be used
+#define NORITAKE_STB 11
+#endif
+#ifndef NORITAKE_SCK
+#warning NORITAKE_SCK not defined, pin 12 will be used
+#define NORITAKE_SCK 12
+#endif
+
+#include <stdarg.h>
+#include <stdio.h>
+#include "ports.h"
+#include "delay.h"
+#include "display.h"
+
+static void write(uint8_t data, uint8_t rs)
+{
+    uint8_t value = 0xf8 + 2 * rs;
+
+    setPinValue(NORITAKE_STB, 0);
+    for (uint8_t i = 0x80; i; i >>= 1)
+    {
+        setPinValue(NORITAKE_SCK, 0);
+        if (value & i)
+            setPinValue(NORITAKE_SIO, 1);
+        else
+            setPinValue(NORITAKE_SIO, 0);
+        setPinValue(NORITAKE_SCK, 1);
+    }
+
+    value = data;
+    for (uint8_t i = 0x80; i; i >>= 1)
+    {
+        setPinValue(NORITAKE_SCK, 0);
+        if (value & i)
+            setPinValue(NORITAKE_SIO, 1);
+        else
+            setPinValue(NORITAKE_SIO, 0);
+        setPinValue(NORITAKE_SCK, 1);
+    }
+    setPinValue(NORITAKE_STB, 1);
+}
+
+static uint8_t read(uint8_t rs)
+{
+    uint8_t data = 0xfc + 2 * rs;
+
+    setPinValue(NORITAKE_STB, PIN_LOW);
+    for (uint8_t i = 0x80; i; i >>= 1)
+    {
+        setPinValue(NORITAKE_SCK, PIN_LOW);
+        if (data & i)
+            setPinValue(NORITAKE_SIO, PIN_HIGH);
+        else
+            setPinValue(NORITAKE_SIO, PIN_LOW);
+        setPinValue(NORITAKE_SCK, PIN_HIGH);
+    }
+
+    setPinMode(NORITAKE_SIO, PIN_INPUT);
+    delayUs(1);
+
+    for (uint8_t i = 0; i < 8; ++i)
+    {
+        setPinValue(NORITAKE_SCK, PIN_LOW);
+        delayUs(1);
+        setPinValue(NORITAKE_SCK, PIN_HIGH);
+
+        data <<= 1;
+        if (getPinValue(NORITAKE_SIO))
+            data |= 1;
+    }
+
+    setPinValue(NORITAKE_STB, PIN_HIGH);
+    setPinMode(NORITAKE_SIO, PIN_OUTPUT);
+    return data;
+}
+
+static void cmd(uint8_t data)
+{
+    delay(54);
+    write(data, 0);
+}
+
+static uint8_t readCmd()
+{
+    uint8_t data = read(0);
+	delayUs(5);
+	return data;
+}
+
+static uint8_t readAddress()
+{
+	return readCmd() & ~0x80;
+}
+
+static void newLine()
+{
+	uint8_t addr = readAddress();
+
+    if (addr < 0x14)
```

Comment:
jak zmienisz kolejnosc warunkow to nie bedziesz musial pisac podwojnych warunkow w else if